### PR TITLE
[extractor/GoogleDrive] Update new Domain/URL for source file download

### PR DESCRIPTION
**Pull request by jazz1611**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

* Update new Domain/URL for source file download
* Identify with the new domain name (example: https://drive.usercontent.google.com/download?id=xxxxxxxx(&export=download))

Fixes https://github.com/yt-dlp/yt-dlp/issues/8976 #

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Review by seproDev — APPROVED** [review]

**Comment by jazz1611** [comment]
Hello @seproDev 

I tried building with this fix and it works stably. Can you apply to yt-dlp officially soon?

```
root@ubuntu:~# ./yt-dlp -Uv "https://drive.usercontent.google.com/download?id=0ByeS4oOUV-49Zzh4R1J6R09zazQ"
[debug] Command-line config: ['-Uv', 'https://drive.usercontent.google.com/download?id=0ByeS4oOUV-49Zzh4R1J6R09zazQ']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.12.30 from yt-dlp/yt-dlp [f10589e34] (linux_exe)
[debug] Python 3.10.12 (CPython x86_64 64bit) - Linux-5.15.0-91-generic-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2023.11.17, mutagen-1.47.0, requests-2.31.0, secretstorage-3.3.1, sqlite3-3.37.2, urllib3-2.1.0, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1798 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2023.12.30 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2023.12.30 from yt-dlp/yt-dlp)
[GoogleDrive] Extracting URL: https://drive.usercontent.google.com/download?id=0ByeS4oOUV-49Zzh4R1J6R09zazQ
[GoogleDrive] 0ByeS4oOUV-49Zzh4R1J6R09zazQ: Downloading video webpage
[GoogleDrive] 0ByeS4oOUV-49Zzh4R1J6R09zazQ: Requesting source file
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 0ByeS4oOUV-49Zzh4R1J6R09zazQ: Downloading 1 format(s): source
[debug] Invoking http downloader on "https://drive.usercontent.google.com/download?id=0ByeS4oOUV-49Zzh4R1J6R09zazQ&export=download&confirm=t"
[download] Destination: Big Buck Bunny.mp4 [0ByeS4oOUV-49Zzh4R1J6R09zazQ].mp4
[download] 100% of  177.36MiB in 00:00:11 at 15.97MiB/s
```

**Comment by seproDev** [comment]
@jazz1611 The extractor will need to be reviewed by another maintainer. Only after they also approve of the changes, can this be merged.

**Review by pukkandan — APPROVED** [review]
Assuming the tests pas, LGTM

**Review by bashonly — APPROVED** [review]

**Pull request merged**
